src/dune_rules/pkg_rules.ml

```diff
@@ -58,6 +58,42 @@ module Pkg_info = struct
   ;;
 end
 
+module Package_universe = struct
+  (* A type of group of packages that are co-installed. Different
+     package universes are unaware of each other. For example the
+     dependencies of the project and the dependencies of one of the dev
+     tools don't need to be mutually co-installable as they are in
+     different universes. *)
+  type t =
+    | Project_dependencies
+    | Dev_tool of Package.Name.t
```

Comment:
By the way, there's still something inconsistent here wrt how contexts are handled. I think the intention is to have a single dev tool for all contexts? If that's the case, why is package resolution taking a context name + `Package_universe.t`? Either dev tools should be context specific, or you should adjust this type to be:

```ocaml
type t =
  | Project_dependencies of Context_name.t
  | Dev_tool of Package.Name.t
```

And remove the context argument in resolve.

Reply:
Ok I just pushed a change that applies your suggestion. Can you let me know if my changes to the `setup_rules` function make sense? I've changed it so that now it only sets up the rules for dev tools if the specified context is the default context, but I don't completely understand this function so I'm not sure if my change is correct.